### PR TITLE
Use xdpyinfo to wait for Xvfb to start up

### PR DESCRIPTION
**Pull request by fhs**
Xvfb sometimes didn't start up within 3 seconds.

### CODE REVIEW

File: fsys_test.go [hunk 1]
```diff
@@ -33,9 +33,18 @@ func TestMain(m *testing.M) {
 				if err := x.Start(); err != nil {
 					log.Fatalf("failed to execute Xvfb: %v", err)
 				}
-				// Give Xvfb some time to start up.
-				// 3 seconds is default for xvfb-run.
-				time.Sleep(3 * time.Second)
+				// Wait for Xvfb to start up.
+				for i := 0; i < 60*60; i++ {
```
Comment (rjkroege) [inline fsys_test.go]: This is an hour if something goes wrong?

File: fsys_test.go [hunk 2]
```diff
@@ -33,9 +33,18 @@ func TestMain(m *testing.M) {
...
+					if err == nil {
+						break
+					}
+					if _, ok := err.(*exec.ExitError); !ok {
+						log.Fatalf("failed to execute xdpyinfo: %v", err)
+					}
+					log.Printf("%v waiting for Xvfb...\n", i)
+					time.Sleep(time.Second)
+				}
 				os.Setenv("DISPLAY", dp)
```
Comment (rjkroege) [inline fsys_test.go]: Isn't it necessary to set the `DISPLAY` in `xdpyinfo`'s environment?

File: fsys_test.go (hunk 1 above)
Comment (rjkroege) [reply fsys_test.go]: Maybe it should be shorter.

Comment (fhs) [reply fsys_test.go]: Yeah, one hour, although public repo jobs can't take longer than 50 minutes: https://docs.travis-ci.com/user/customizing-the-build/#build-timeouts

I don't know how Travis schedules jobs. I was kind of assuming jobs can be preempted for higher priority jobs. I can decrease the time. 5 minutes maybe?

File: fsys_test.go (hunk 2 above)
Comment (fhs) [reply fsys_test.go]: Nope, I'm passing the display explicitly with the `-display` flag.

File: fsys_test.go (hunk 1 above)
Comment (rjkroege) [reply fsys_test.go]: 5 minutes sounds good to me. If it's not going to start in 5 minutes, we have a problem.

File: fsys_test.go (hunk 2 above)
Comment (rjkroege) [reply fsys_test.go]: Indeed. Sorry, I should have read more carefully.

File: fsys_test.go (hunk 1 above)
Comment (fhs) [reply fsys_test.go]: I've changed it to 5 seconds.

**Comment by rjkroege** [comment]
thanks.

**Pull request merged**